### PR TITLE
feat(webhook): Support multiple arrays of tags is matchers.

### PR DESCRIPTION
**Pull request by npalm**
Refactored the matcher to support multiple arrays of tags. This allow to match multiple set of tags on a single runner configuration. Useful to match for example `[self-hosted, ubuntu-latest]` and `[self-hosted, ubuntu-2204]` on the same runner.

### CODE REVIEW

**Comment by npalm** [comment]
close #2715

**Comment by GuptaNavdeep1983** [comment]
@npalm, not sure if this PR gives an error on applying without commit https://github.com/philips-labs/terraform-aws-github-runner/pull/2736/commits/2ab5f465796c66bff1ccddb83471d62b79c1ae27#diff-69a0e0ca008de526fb8fe2c1d1995d6632c16a9cd1bc58c9f58eec9ce1676f30!

**Comment by npalm** [comment]
Tested the branch with the example for multi-runners, all works fine.

File: modules/webhook/lambdas/webhook/src/webhook/handler.test.ts [hunk 1]
```diff
@@ -418,6 +418,48 @@ describe('handler', () => {
     });
   });
 
+  it('Check webhook will accept jobs when matchers accepts multiple labels.', async () => {
+    process.env.RUNNER_CONFIG = JSON.stringify([
+      {
+        ...queuesConfig[0],
+        matcherConfig: {
+          labelMatchers: [['self-hosted', 'arm64', 'linux', 'ubuntu-latest']],
```
Comment (GuptaNavdeep1983) [inline modules/webhook/lambdas/webhook/src/webhook/handler.test.ts:426]: should we have multiple labels supported by this runner to let the test be more effective?

File: modules/webhook/lambdas/webhook/src/webhook/handler.ts [hunk 2]
```diff
@@ -161,18 +161,22 @@ function isRepoNotAllowed(repoFullName: string, repositoryWhiteList: string[]):
...
+  runnerLabelsMatchers: string[][],
+  workflowLabelCheckAll: boolean,
+): boolean {
+  runnerLabelsMatchers = runnerLabelsMatchers.map((runnerLabel) => {
+    return runnerLabel.map((label) => label.toLowerCase());
   });
   const match = workflowLabelCheckAll
-    ? workflowJobLabels.every((l) => runnerLabels.includes(l.toLowerCase()))
-    : workflowJobLabels.some((l) => runnerLabels.includes(l.toLowerCase()));
+    ? workflowJobLabels.every((wl) => runnerLabelsMatchers.some((rl) => rl.includes(wl.toLowerCase())))
```
Comment (GuptaNavdeep1983) [inline modules/webhook/lambdas/webhook/src/webhook/handler.ts:173]: Quite complicated query.

File: modules/webhook/lambdas/webhook/src/webhook/handler.test.ts (hunk 1 above)
Comment (npalm) [reply modules/webhook/lambdas/webhook/src/webhook/handler.test.ts]: Good catch

File: modules/webhook/lambdas/webhook/src/webhook/handler.ts (hunk 2 above)
Comment (npalm) [reply modules/webhook/lambdas/webhook/src/webhook/handler.ts:173]: I think we need this, to be able to support cases like ubuntu-latest and ubuntu-2204. Alternative is deploying another stack, wich is even worse in my point of view

**Review by GuptaNavdeep1983 — APPROVED** [review]

**Pull request merged**